sdk/search/Azure.Search.Documents/src/Indexes/Models/AnalyzeTextOptions.cs

```diff
@@ -40,18 +40,37 @@ public AnalyzeTextOptions(string text, LexicalTokenizerName tokenizerName)
             CharFilters = new ChangeTrackingList<string>();
         }
 
-        /// <summary> The name of the analyzer to use to break the given text. If this parameter is not specified, you must specify a tokenizer instead. The tokenizer and analyzer parameters are mutually exclusive. </summary>
+        /// <summary>
+        /// Initializes a new instance of AnalyzeRequest.
+        /// </summary>
+        /// <param name="text">Required text to break into tokens.</param>
+        /// <param name="normalizerName">The name of the tokenizer to use to break the given <paramref name="text"/>.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="text"/> is null.</exception>
+        public AnalyzeTextOptions(string text, LexicalNormalizerName normalizerName)
+        {
+            Text = text ?? throw new ArgumentNullException(nameof(text));
+            NormalizerName = normalizerName;
+
+            TokenFilters = new ChangeTrackingList<TokenFilterName>();
+            CharFilters = new ChangeTrackingList<string>();
```

Comment:
Should we add an internal parameterless .ctor to do all of this stuff and let all the others call `: this()`?

Reply:
We already have such a ctor. I changed the code to call it.